test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -164,21 +179,47 @@ func (r *DockerMachinePoolReconciler) SetupWithManager(ctx context.Context, mgr
 			builder.WithPredicates(
 				predicates.ClusterUnpausedAndInfrastructureReady(ctrl.LoggerFrom(ctx)),
 			),
-		).Complete(r)
+		).Build(r)
 	if err != nil {
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
+
+	r.controller = c
+	r.recorder = mgr.GetEventRecorderFor("dockermachinepool-controller")
+	r.externalTracker = external.ObjectTracker{
+		Controller: c,
+		Cache:      mgr.GetCache(),
+	}
+
 	return nil
 }
 
-func (r *DockerMachinePoolReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool) error {
-	pool, err := docker.NewNodePool(ctx, r.Client, cluster, machinePool, dockerMachinePool)
+func (r *DockerMachinePoolReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, dockerMachinePool *infraexpv1.DockerMachinePool) error {
+	_ = ctrl.LoggerFrom(ctx)
+
+	labelFilters := map[string]string{dockerMachinePoolLabel: dockerMachinePool.Name}
+	machines, err := docker.ListMachinesByCluster(ctx, cluster, labelFilters)
 	if err != nil {
-		return errors.Wrap(err, "failed to build new node pool")
+		return errors.Wrapf(err, "failed to list all machines in the cluster")
 	}
 
-	if err := pool.Delete(ctx); err != nil {
-		return errors.Wrap(err, "failed to delete all machines in the node pool")
+	// Iterate over the Docker containers directly instead of DockerMachines to ensure none are left behind.
```

Comment:
Yes, in this case the "machines" are actually the containers. I agree it's awkward since the terms is so overloaded, but I'll change it to something like `externalMachines` or `containers`.